src/planner/vectorfield/MoveEndEffectorOffsetVectorField.cpp

```diff
@@ -4,144 +4,98 @@
-  if (mLinearTolerance < 0)
-    throw std::invalid_argument("Linear tolerance is negative");
-  if (mAngularGain < 0)
-    throw std::invalid_argument("Angular gain is negative");
+  if (mMinDistance < 0)
+    throw std::invalid_argument("Minimum distance must be non-negative.");
+  if (mDirection.norm() == 0)
+    throw std::invalid_argument("Direction must be non-zero");
+  if (mMaxDistance < mMinDistance)
+    throw std::invalid_argument("Max distance is less than minimum distance.");
+  if (mPositionTolerance < 0)
+    throw std::invalid_argument("Position tolerance is negative");
   if (mAngularTolerance < 0)
     throw std::invalid_argument("Angular tolerance is negative");
-  if (mOptimizationTolerance < 0)
-    throw std::invalid_argument("Optimization tolerance is negative");
 
-  mTargetPose = mStartPose;
-  mTargetPose.translation() += mVelocity * (mEndTime - mStartTime);
-  mEndTime += _padding;
+  // Normalize the direction vector
+  mDirection.normalize();
 }
 
 //==============================================================================
-bool MoveEndEffectorOffsetVectorField::operator()(
-    const aikido::statespace::dart::MetaSkeletonStateSpacePtr& _stateSpace,
-    double,
-    Eigen::VectorXd& _dq)
+bool MoveEndEffectorOffsetVectorField::evaluateCartesianVelocity(
+    const Eigen::Isometry3d& pose, Eigen::Vector6d& cartesianVelocity) const
 {
-  using Eigen::Isometry3d;
-  using Eigen::Vector3d;
-  using Eigen::Vector6d;
-  using dart::math::logMap;
-
-  const Isometry3d currentPose = mBodynode->getTransform();
-  Vector3d currentWorkspacePose = currentPose.translation();
-  Vector3d targetWorkspacePose = mTargetPose.translation();
-
-  // Compute the "feed-forward" term necessary to move at a constant velocity
-  // from the start to the goal.
-  const Vector3d& linearFeedforward = mVelocity;
+  using aikido::planner::vectorfield::computeGeodesicError;
 
-  // Compute positional error orthogonal to the direction of motion by
-  // projecting out the component of the error along that direction.
-  const Vector3d linearError = targetWorkspacePose - currentWorkspacePose;
-  const Vector3d linearOrthogonalError
-      = linearError - linearError.dot(mLinearDirection) * mLinearDirection;
-  // Compute rotational error.
-  const Vector3d rotationError
-      = logMap(mTargetPose.rotation().inverse() * currentPose.rotation());
-
-  // Compute the desired twist using a proportional controller.
-  Vector6d desiredTwist;
-  desiredTwist.head<3>() = mAngularGain * rotationError;
-  desiredTwist.tail<3>()
-      = linearFeedforward + mLinearGain * linearOrthogonalError;
-
-  bool result = computeJointVelocityFromTwist(
-      desiredTwist,
-      _stateSpace,
-      mBodynode,
-      mOptimizationTolerance,
-      mTimestep,
-      mPadding,
-      _dq);
-  return result;
+  Eigen::Vector6d desiredTwist = computeGeodesicTwist(pose, mStartPose);
+  desiredTwist.tail<3>() = mDirection;
+  cartesianVelocity = desiredTwist;
```

Comment:
Yes. This is removed when moving from `muul` design to `prpy` design.
In `muul`, `MoveHandStraightVectorField` defines a timed workspace path by a distance and linear velocity, which is similar with the requirement of `PlanWorkspacePath()` in `prpy`, so linear gain and rotation gain are used.
Now `planToEndEffectorOffset` moves a range [minimum distance, maximum distance). It is not easy to estimate rotational error and linear orthogonal error, because no goal pose is known.
I guess that is the reason that no control gain is used in `planToEndEffectorOffset()` of `prpy`.

In a coming `PlanWorkspacePath` vector field, linear and rotation control gains are used in correcting the deviation from a reference workspace path. 

Does this make sense?

Reply:
Yes, that makes sense. 👍 Thanks for the explanation.